File: binary_search.cs
using System;
					
public class Program
{
	public static void Main()
	{        
		var els = new int[] {2, 4, 5, 6, 9, 24, 25, 48, 59, 100};

		BinarySearch(els, 0);
		BinarySearch(els, -100);
		BinarySearch(els, 20);

		BinarySearch(els, 24);
		BinarySearch(els, 5);
		BinarySearch(els, 9);

		BinarySearch(els, 100);
		BinarySearch(els, 2);
	}
		
	public static int BinarySearch(int[] elements, int number)
	{
		var (left, right) = (0, elements.Length - 1);

		if (elements[left] == number) 
		{
			Console.WriteLine($"Found the number {number} at index {left}");
			return left;
		}
		else if (elements[right] == number)
		{
			Console.WriteLine($"Found the number {number} at index {right}");
			return right;
		}

		while (left < right)
		{
			var middle_index = left + (right - left);
			if (elements[middle_index] == number)
			{
				Console.WriteLine($"Found the number {number} at index {middle_index}");
				return middle_index;
			}
			else if (number < elements[middle_index])
			{
				right = middle_index - 1;
			}
			else if (number > elements[middle_index])
			{
				left = middle_index + 1;
			}
		}
		Console.WriteLine($"Number {number} not found in the array");
		return -1;
	}
}

File: binary_search.py
def binary_search(elements, number):
    left, right = 0, len(elements)-1

    if elements[left] == number: 
        print(f'found the number {number} at index {left}')
        return left
    elif elements[right] == number:
        print(f'found the number {number} at index {right}')
        return right

    while left < right:
        middle_index = left + (right - left) // 2
        if elements[middle_index] == number:
            print(f'found the number {number} at index {middle_index}')            
            return middle_index
        elif number < elements[middle_index]:
            right = middle_index - 1
        elif number > elements[middle_index]:
            left = middle_index + 1
    
    print(f'number {number} not found in the array')
    return -1

els = [2, 4, 5, 6, 9, 24, 25, 48, 59, 100]

binary_search(els, 0)
binary_search(els, -100)
binary_search(els, 20)

binary_search(els, 24)
binary_search(els, 5)
binary_search(els, 9)

binary_search(els, 100)
binary_search(els, 2)
